### PR TITLE
Jochem

### PR DESCRIPTION
**Pull request by YousefED**

### CODE REVIEW

File: server/players.ts [hunk 1]
```diff
@@ -1,5 +1,17 @@
 import { Kit, Player } from "../shared/sharedModels";
 
+type PlayerTeamKit = "TLJ" | "BOA" | "FDJ" | "EUC" | "MOV" | "COF" | "TFR";
```
Comment (YousefED) [inline server/players.ts]: why is this here?

File: src/index.ts [hunk 2]
```diff
@@ -2,13 +2,19 @@ import "./style/css/style.css";
...
 
 async function getPlayersFromServer() {
-    const players = await axios.get<Player[]>("http://localhost:4000/players");
-    const positions = await axios.get<
+    const playerRequest = axios.get<Player[]>("http://localhost:4000/players");
+    const positionRequest = axios.get<
         Array<{ name: string; position: number }>
     >("http://localhost:4000/positions");
 
+    const [players, positions] = await Promise.all([
```
Comment (YousefED) [inline src/index.ts]: nice :) this was it indeed

File: src/util/freePosition.ts [hunk 3]
```diff
@@ -28,10 +63,55 @@ export function getClosestFreePosition(
...
+
+    // Sort the other boxes based on proximity.
+    const closest = _boxes.sort(
+        (a, b) => distToDesired(boxToPlace, a) - distToDesired(boxToPlace, b)
+    );
+
+    // Now for closest neighbors check the surrounding positions and select a position.
+    for (const closebox of closest) {
+        // Check all positions and
+        const candidates = [
```
Comment (YousefED) [inline src/util/freePosition.ts]: shouldn't there be 4 options?

File: src/util/freePosition.ts [hunk 4]
```diff
@@ -28,10 +63,55 @@ export function getClosestFreePosition(
...
+            { ...closebox, y: closebox.y - closebox.height },
+            { ...closebox, x: closebox.x - closebox.width },
```
Comment (YousefED) [inline src/util/freePosition.ts]: this should probably be  - boxToPlace.width, and in line 85 - boxToPlace.height

File: server/players.ts (hunk 1 above)
Comment (jochem725) [reply server/players.ts]: That’s a good question
